lib/helper/generatorHelper.py

```diff
@@ -293,66 +281,26 @@ def calculate(env, crop_data=None, calc_si=0, waveform=0, revision="", pkg_platf
-            sample_data = {
-                'sample_dp': env.img_sample_dp,
-                'configuration': {
-                    'generator': {
-                        DEFAULT_DCTINPUTLATENCY_GENERATOR_NAME: {
-                            'path': 'lib.generator.dctInputLatencyGenerator'
-                        }
-                    }
-                }
-            }
-        elif waveform == 2:
-            # AIL, dctAnimationInputLatencyGenerator
-            sample_data = {
-                'sample_dp': env.img_sample_dp,
-                'configuration': {
-                    'generator': {
-                        DEFAULT_DCTANIMATIONINPUTLATENCY_GENERATOR_NAME: {
-                            'path': 'lib.generator.dctAnimationInputLatencyGenerator'
-                        }
-                    }
-                }
-            }
-        elif waveform == 3:
-            # FT, dctFrameThroughputGenerator
-            current_sample_converter_name = FRAME_THROUGHPUT_SAMPLE_CONVERTER_NAME
-            sample_settings = copy.deepcopy(FT_SAMPLE_CONVERTER_SETTINGS)
-            sample_data = {
-                'sample_dp': env.img_sample_dp,
-                'configuration': {
-                    'generator': {
-                        DEFAULT_DCTFRAMETHROUGHPUT_GENERATOR_NAME: {
-                            'path': 'lib.generator.dctFrameThroughputGenerator'}}},
-                'orig_sample': env.img_output_sample_1_fn}
-        else:
-            # normal cases
-            sample_data = {'sample_dp': env.img_sample_dp,
-                           'configuration': {'generator': {
-                               DEFAULT_DCTRUNTIME_GENERATOR_NAME: {'path': 'lib.generator.dctRunTimeGenerator'}}}}
-            if calc_si == 1:
-                sample_data['configuration']['generator'][DEFAULT_SPEEDINDEX_GENERATOR_NAME] = {
-                    'path': 'lib.generator.speedIndexGenerator'}
+            index_config['image-converter-name']: {'video_fp': env.video_output_fp, 'output_img_dp': env.img_output_dp,
+                                                   'convert_fmt': index_config['image-converter-format'],
+                                                   'current_fps': validate_result[DEFAULT_FPS_VALIDATOR_NAME][
+                                                       'output_result'],
+                                                   'exec_timestamp_list': exec_timestamp_list}}
+        converter_result = run_modules(converter_settings, converter_data[index_config['image-converter-name']])
+
+        sample_settings = {'modules': {index_config['sample-converter-name']: {'path': index_config['sample-converter-path']}}}
+        sample_data = {'sample_dp': env.img_sample_dp,
+                       'configuration': {'generator': {
+                           index_config['module-name']: {'path': index_config['module-path']}}}}
 
         # {1:{'fp': 'xxcxxxx', 'DctRunTimeGenerator': 'dctobj', 'SSIMRunTimeGenerator': None, },
```

Comment:
do you also want to change the commented lines here? DctRunTimeGenerator -> RunTimeDctGenerator